File: binV2/third_problem.rb
require 'json'

n = ARGV[1].to_i
n = 1 if n.zero?

input_data = [
  { "name": "Apples", "quantity_sold": 120 },
  { "name": "Oranges", "quantity_sold": 75 },
  { "name": "Bananas", "quantity_sold": 200 }
]

sorted_data = input_data.sort_by { |data| data[:quantity_sold] }.reverse!

top_products = sorted_data.take(n)

print JSON.pretty_generate({ data: top_products })

File: binsV1/binV12/third_problem.js
const n = Number(process.argv[3]) || 2;

const products = [
  { name: "Apples", quantity_sold: 120 },
  { name: "Oranges", quantity_sold: 75 },
  { name: "Bananas", quantity_sold: 200 },
];

const topProducts = products
  .toSorted((a, b) => b.quantity_sold - a.quantity_sold)
  .slice(0, n);

console.log(JSON.stringify({ data: topProducts }, null, 2));
